### PR TITLE
[Form Field v2] Checkbox and Radiobutton support

### PR DESCRIPTION
**Pull request by libertymayc**
Checkbox and RadioButton integration with FF full support
Part of V2 Form Field features

**PENDING**: Warning and Readonly support (features missing & both needed in CB/RB)

### CODE REVIEW

File: packages/core/stories/form-field/form-field.stories.tsx [hunk 1]
```diff
@@ -205,6 +205,16 @@ export const WithValidation: ComponentStory<typeof FormField> = (props) => {
...
+      <FormField validationStatus="error" {...props}>
+        <FormLabel>Error Form Field</FormLabel>
+        <Checkbox label="Option 1" />
+        <FormHelperText>Helper text</FormHelperText>
+      </FormField>
+      <FormField validationStatus="error" {...props}>
+        <FormLabel>Error Form Field</FormLabel>
+        <RadioButton label="Option 1" />
+        <FormHelperText>Helper text</FormHelperText>
+      </FormField>
```
Comment (alycrys) [inline packages/core/stories/form-field/form-field.stories.tsx:217]: I think design can give some more realistic examples to add here, based on their research on forms

Comment (libertymayc) [reply packages/core/stories/form-field/form-field.stories.tsx:217]: agreed, waiting on specs from @bhoppers2008

**Comment by alycrys** [comment]
There is a scenario that is missing, the `readOnly` form field. There was a discussion about read-only states on checkbox and radio button. Not sure what the status is on that. Maybe @navkaur76 or @bhoppers2008 can advise on this?

**Comment by navkaur76** [comment]
> There is a scenario that is missing, the `readOnly` form field. There was a discussion about read-only states on checkbox and radio button. Not sure what the status is on that. Maybe @navkaur76 or @bhoppers2008 can advise on this?

For CB/RB the read-only state has been designed, and as far as I'm aware can be added to the component. It was left as an agenda item for Q2 as there was some ADA questions around it to discuss with Stella.

**Comment by libertymayc** [comment]
> > There is a scenario that is missing, the `readOnly` form field. There was a discussion about read-only states on checkbox and radio button. Not sure what the status is on that. Maybe @navkaur76 or @bhoppers2008 can advise on this?
> 
> For CB/RB the read-only state has been designed, and as far as I'm aware can be added to the component. It was left as an agenda item for Q2 as there was some ADA questions around it to discuss with Stella.

it's tracked by https://github.com/jpmorganchase/salt-ds/issues/1567

File: packages/core/src/radio-button/RadioButton.tsx [hunk 2]
```diff
@@ -90,6 +91,24 @@ export const RadioButton = forwardRef<HTMLLabelElement, RadioButtonProps>(
...
+      a11yProps: {
+        "aria-describedby": formFieldDescribedBy,
+        "aria-labelledby": formFieldLabelledBy,
+      } = {},
+      disabled: formFieldDisabled,
+      validationStatus: formFieldValidationStatus,
+    } = useFormFieldProps();
+
+    const disabled = formFieldDisabled ?? disabledProp;
+    const error = formFieldValidationStatus === "error" ?? errorProp;
```
Comment (alycrys) [inline packages/core/src/radio-button/RadioButton.tsx:110]: i think this should be done on the RadioButtonGroup. it's unlikely the radio button will be used directly inside the FF

Comment (libertymayc) [reply packages/core/src/radio-button/RadioButton.tsx:110]: updated

**Review by alycrys — APPROVED** [review]

**Pull request merged**